fili-core/src/main/java/com/yahoo/bard/webservice/web/filters/ApiFilters.java

```diff
@@ -28,4 +29,41 @@ public ApiFilters() {
     public ApiFilters(Map<Dimension, Set<ApiFilter>> filters) {
         super(filters);
     }
+
+    /**
+     * Merges two ApiFilters.
+     *
+     * e.g.
+     * ApiFilters f1 = {
+     *      dim1: [apiFilter1, apiFilter2]
+     *      dim2: [apiFilter3]
+     * }
+     * ApiFilters f2 = {
+     *      dim2: [apiFilter3, apiFilter4]
+     *      dim3: [apiFilter5]
+     * }
+     *
+     * result = {
+     *      dim1: [apiFilter1, apiFilter2]
+     *      dim2: [apiFilter3, apiFilter4]
```

Comment:
This seems wrong.  If I'm filtering to apiFilter3, the narrower filter not the broader filter should apply.  You can do this by just anding them together rather than merging them.

Reply:
I'm not sure I totally understand. in this situation I meant for apiFilter3 in both f1 and f2 to be the exact same filter. I think the example you just gave would be trying to merge the filters

dim1|id-in[val1]
and 
dim1|id-in[val2,val3]

I was under the impression that both of those filters should be used and that the druid filter builder would handle merging the logic. In that case the result be something like:

{ dim1: [ dim1|id-in[val1], dim1|id-in[val2,val3] ] as Set } as Map

Reply:
I guess that's more of a union than a merge though.